Mathlib/Probability/Kernel/Condexp.lean

```diff
@@ -148,9 +157,24 @@ theorem integrable_toReal_condexpKernel (hm : m ≤ mΩ) {s : Set Ω} (hs : Meas
 
 end Integrability
 
+lemma condexpKernel_ae_eq_condexp [IsFiniteMeasure μ]
+    (hm : m ≤ mΩ) {s : Set Ω} (hs : MeasurableSet s) :
```

Comment:
Thinking back about this, we definitely shouldn't change `Measure.trim`, as we really want to have a measure on the sigma-algebra `m` by definition, not on `m ⊓ m0` (which might coincide with `m` in good situations, but not definitionally).

Reply:
I changed the definition of `condexpKernel` to use the inf. It indeed saves many `m ≤ mΩ` assumptions and I don't see any drawback. I added a new lemma `AEStronglyMeasurable'.mono` to help in a proof.

It might be that changing `condexp` in the same way would also remove some assumptions, but that is a big change impacting many files and I don't have the courage to try that now.

I agree that we don't want to do that for `Measure.trim`.